MERLin/MERLin/Events/EventsListening.swift

```diff
@@ -19,13 +19,13 @@ public protocol AnyEventsListening: class {
 
 public protocol EventsListening: AnyEventsListening {
     associatedtype EventsType: EventProtocol
-    @discardableResult func registerToEvents(for producer: AnyEventsProducer, events: EventsProxy<EventsType>) -> Bool
+    @discardableResult func registerToEvents(for producer: AnyEventsProducer, events: Observable<EventsType>) -> Bool
```

Comment:
Do you need the `events` parameter hear anymore? The Producer seems to do everything that events used to do.

Reply:
events is needed for generics. There is no way to cast to AnyEvents of a specific Events Type.

Reply:
See it as a needed constraint, to avoid the listener to be called for not interesting events.
You could still trick the listener by passing a producer that does not produces the specified events, but if you leave it to MERLin, the right thing will be done.